### PR TITLE
Update Gubbins to v2.3.1

### PR DESCRIPTION
**Pull request by andrewjpage**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

It fails linting on the setup_py_install_args check, however if you add the suggested code the build fails.

### CODE REVIEW

File: recipes/gubbins/meta.yaml [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
+{% set version = "2.3.1" %}
+{% set md5 = "157e91666afc6160b4fe15327dcd375e" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 build:
-  number: 0
+  number: 1
```
Comment (druvus) [inline recipes/gubbins/meta.yaml]: Keep 0 for new versions.

Comment (andrewjpage) [reply recipes/gubbins/meta.yaml]: Travis fell over with a bioconda error, totally unrelated to this, but forcing another build fixed it.

Comment (bgruening) [reply recipes/gubbins/meta.yaml]: Andrew do you have the error? It is still failing.

**Comment by druvus** [comment]
Added in #6516

**Pull request closed without merging**